### ISSUE TITLE
elementum 0.1.50 cannot add torrent

### ISSUE THREAD

**Issue by std781** [elgatito/plugin.video.elementum#565]
Hi, 
I have an issue with Android TV 9. 
Latest Kodi -18.6-Leia-armeabi-v7a
Plugin starts but after i choose add torrent nothing happens.
Attaching log here.
[kodi.zip](https://github.com/elgatito/plugin.video.elementum/files/4434218/kodi.zip)

**Comment by burekas7** [comment]
I have a similar issue but with a season torrent.
I have a season torrent file, when I'm trying to open it in order to add a torrent manually.
But something is unstable, from ~15 tries, it only once shows the dialog with the torrents that included in the file.
(Just for referrence I tried it with Quasar addon, and there it's ok)

Kodi debug log:
https://paste.ubuntu.com/p/S6FF4JzG4C

Elementum 0.1.50
Kodi 17.6
Skin: Titan
Windows 10, 64bit

**Comment by elgatito** [comment]
@burekas7 your log shows nothing. Something is added in the very end and added fine, as for the log.

**Comment by elgatito** [comment]
@std781 how you add the torrent? via Kodi - Elementum - Add torrent ? Or via web? What you want to play there?

**Comment by burekas7** [comment]
> @burekas7 your log shows nothing. Something is added in the very end and added fine, as for the log.
@elgatito 

For example, trying to do "Add Torrent" with this file:
(You can download here - Season torrent)
https://f2h.io/dck5qpxo68m1

In this log: https://paste.ubuntu.com/p/9kfJYbhtGR
On the first time nothing happened, but on the first time it worked and show the dialog as expected.
I've also noticed that when I press ESC immediately after pressing OK, the dialogs shows and it works everytime, but it also quit from the addon, you can see this in this log: https://paste.ubuntu.com/p/ZxKM5ZYK2N

It seems to be maybe something with timings.

![image](https://user-images.githubusercontent.com/13424257/78894772-61936280-7a76-11ea-8f1e-a1d68ccce775.png)

![image](https://user-images.githubusercontent.com/13424257/78893057-41ae6f80-7a73-11ea-9f87-43714e66726a.png)

---
And by the way, why when I press on "Cancel" it keeps asking "Keep Downloading"?
I don't start the downloading.
The log has this issue:
https://paste.ubuntu.com/p/ZxKM5ZYK2N
![image](https://user-images.githubusercontent.com/13424257/78894716-41fc3a00-7a76-11ea-8d6a-6d0c1f6f79a9.png)

**Comment by elgatito** [comment]
@burekas7 I cannot see anything wrong in the log file. In both logs file selector is opened and then have no respond from user. 
I have tried a lot of times (via elementum->add torrent and "web - > play") and always got a file selector window.

Your Kodi says it is opening the file select window. But I can't see anything after that.

**Comment by elgatito** [comment]
@burekas7 please, try with 0.1.51.

**Comment by burekas7** [comment]
No.
Still doesn't work and even worse.
In 0.1.50, sometimes it works and sometimes not.
With 0.1.51 non of the tries works, even with the ESC key trick.

Log: https://paste.ubuntu.com/p/djDSBbrmbp

**Comment by elgatito** [comment]
@burekas7 I could not make it to fail adding a torrent even once. It's always adding. I've tried with magnet/torrents, tried to add from website, tried with plugin "Add torrent" menu. Always added.
I do not know what you do, taking in mind your log is silent about Elementum errors or even messages. 
I would try with empty portable Kodi. Maybe some other plugin is breaking something.

**Comment by burekas7** [comment]
I will try to make cleam kodi version to check it, there few options:
- Kodi version
- Skin
- Python issue
- Addons

I don't know why pressing the 'escape' key right after pressing the "ok" button in the dialog make it works, the problem is that it also makes the addon to quit and goes out to the general "videos" kodi screen.
The 'escape' trick also helps sometimes to the "search" issues that I have with Elementum (which I opened in another issue):
- Right after typing and confirm the text for search and confirm in the keyboard screen
- Or after selecting one of the result to start playing.
If I don't pressing the 'escape' key nothing would happen, it kind of "release" something or there is race condition which makes it unstable.
I don't have it in another addon or Quasar which is the most similar.

I will try to makes tests and show you videos of what I meant.
Thanks anyway.
@elgatito

**Comment by elgatito** [comment]
@burekas7  I would first try to disable 'redalert' plugin. It fires every time something happen in GUI.

I don't understand, you don't see file selector when you add a torrent? And select a file to download?

Because in the log first time I see there is a torrent insert action, where you manually select a torrent file. And then nothing happens, it does not go back to elementum.

**Comment by elgatito** [comment]
@burekas7  Please, try with the latest 0.1.52 version and post a new log.

**Comment by burekas7** [comment]
@elgatito 
Thanks, it seems to be much better and stable with 0.1.52

I don't know about @std781 but I think it can be closed for now.

**Comment by elgatito** [comment]
@burekas7 it should not be better, because nothing was made with that, only added log messages to identify the problem.

**Comment by burekas7** [comment]
So I have not catched it yet again to post you a new log with the new log messages.
I have tried several times as before and all the tries were ok, but it was unstable issue (sometimes it worked and sometimes not) so it probably fell into the good stats, if I will catch it again I will report.

Thanks.

**Comment by elgatito** [comment]
@burekas7 @std781 
Please, try with Elementum 0.1.56.

**Comment by abugamel** [comment]
Hi guys

I added torrent link by yatse app on Android phone then i got crashed on elementum

Still have a message" elementum has crashed. Restarting....

Please help 

@elgatito

Kodi log


https://drive.google.com/file/d/1JD35ScLb8jRsaheqbIq1pyRCARgKCuOo/view?usp=drivesdk

**Comment by elgatito** [comment]
@burekas7 @std781 Please, try with 0.1.62 and further.

**Comment by burekas7** [comment]
@elgatito 
Tested with 0.1.62 and it seems to be ok, the list is opening and the Cancel button close it.
Thanks.
